src/plugins/intel_cpu/tests/functional/single_layer_tests/matmul.cpp

```diff
@@ -182,9 +182,29 @@ TEST_P(MatMulLayerCPUTest, CompareWithRefs) {
             }
         }
     }
-
     run();
-    CheckPluginRelatedResults(compiledModel, cpuNodeType);
+    if (compiledModel) {
+        auto execGraph = compiledModel.get_runtime_model();
+        bool useMlas = false;
+        for (const auto &node : execGraph->get_ops()) {
+            const auto & rtInfo = node->get_rt_info();
+            auto getExecValue = [&rtInfo](const std::string & paramName) -> std::string {
+                auto it = rtInfo.find(paramName);
+                IE_ASSERT(rtInfo.end() != it);
+                return it->second.as<std::string>();
+            };
+
+            if (getExecValue(ExecGraphInfoSerialization::LAYER_TYPE) == "FullyConnected") {
+                auto primType = getExecValue(ExecGraphInfoSerialization::IMPL_TYPE);
+                if (primType.find("gemm_mlas") != std::string::npos) {
+                    useMlas = true;
+                }
+            }
+        }
+        if (!useMlas) {
+            CheckPluginRelatedResults(compiledModel, cpuNodeType);
+        }
+    }
```

Comment:
Is not it better to simply set "gemm_mlas" as an expected primitive type in some version of `filterFusingParams` but used only for fullyConnected test instances? The same approach as the special postops filtering you've introduced.